micro-rdk/src/esp32/conn/network.rs

```diff
@@ -609,6 +609,12 @@ impl Esp32ExternallyManagedNetwork {
             && ev_id == esp_idf_svc::sys::wifi_event_t_WIFI_EVENT_STA_DISCONNECTED
         {
             data.connected.store(false, Ordering::Release);
+            if !ev_data.is_null() {
+                let disconn: *const esp_idf_svc::sys::wifi_event_sta_disconnected_t =
+                    unsafe { core::mem::transmute(ev_data) };
+                let reason: super::wifi_error::WifiErrReason = (*disconn).reason.into();
+                log::error!("sta disconnected: {:?}", reason);
```

Comment:
I believe that the intention of this ticket (which was admittedly light on details) was to actually inspect the internal structure of the `StaConnected` and `StaDisconnected` events up around line 200 and report interesting context around these events (probably disconnection is generally more interesting).

Reply:
that's probably a better place than where I found originally, yeah.
I shifted the logging to there if you'd like to take a look.

So does that mean we want to look deeper into `StaConnected` as well for this ticket? It seems we would be able to log the ssid but not much else, https://docs.esp-rs.org/esp-idf-svc/esp_idf_svc/wifi/struct.StaConnectedRef.html